module5/src/main/java/Main.java

```diff
@@ -0,0 +1,182 @@
+import bean.Faculty;
+import bean.Group;
+import bean.Student;
+import exceptions.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Main {
+    public static void main(String[] args) {
+        List<Faculty> facultyList = createFacultyList();
+        try {
+            validateFacultyGroupStudentsSubjectIsEmpty(facultyList);
+            calculateAverageGradeForEachStudent(facultyList);
+            System.out.println();
+            calculateAverageGradeInOneGroup(facultyList, "Math", 1, "English");
+            System.out.println();
+            calculateAverageGradeForOneSubjectInUniversity(facultyList, "Math");
+        } catch (MyException e) {
+            e.getStackTrace();
+        }
+    }
+
+    /**
+     * Validation what this data is not null.
+     *
+     * @param facultyList all faculty in university.
+     * @throws MyException parent class for all exceptions in this app.
+     */
+    private static void validateFacultyGroupStudentsSubjectIsEmpty(List<Faculty> facultyList) throws MyException {
+        if (facultyList.isEmpty()) {
+            throw new EmptyFacultyListException("There is no faculties in university.");
+        }
+        for (Faculty faculty : facultyList) {
+            if (faculty.getGroups().isEmpty()) {
+                throw new EmptyFacultyGroupsException("There is no group in " + faculty.getName() + " faculty");
+            }
+            for (Group group : faculty.getGroups()) {
+                if (group.getStudentList().isEmpty()) {
+                    throw new EmptyStudentListException("There is no students in " + group.getName() + " group.");
+                }
+                for (Student students : group.getStudentList()) {
+                    if (students.getSubjects().isEmpty()) {
+                        throw new EmptySubjectException("There is " + students.getName() + " no subjects");
```

Comment:
Sorry, `There is a student ... with no subjects`
or `Student ... has no subjects`